src/test/java/org/mule/test/extension/file/common/api/FileTestHarness.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.extension.file.common.api;
+
+import org.mule.functional.junit4.rules.ExpectedError;
+
+import java.nio.file.Paths;
+
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestRule;
+
+/**
+ * A {@link TestRule} which provides tools and environment for reusing the same test regardless of the remote server type (FTP,
+ * SFTP, etc) or the authentication method
+ *
+ * @since 1.0
+ */
+public interface FileTestHarness extends TestRule {
+
+  String HELLO_WORLD = "Hello World!";
+  String HELLO_FILE_NAME = "hello.json";
+  String BINARY_FILE_NAME = "binary.bin";
+  String HELLO_PATH = Paths.get("files", HELLO_FILE_NAME).toString();
+  String DEFAULT_FTP_HOST = "localhost";
+  String FTP_SERVER_BASE_DIR = Paths.get("target", "ftpserver").toString();
+  String WORKING_DIR_SYSTEM_PROPERTY = "workingDir";
+  String WORKING_DIR = "base";
+
+  /**
+   * Creates a test hello world file
+   * @throws Exception if the method fails
```

Comment:
@throws Exception if the method fails > thanks for adding a so useful description

Reply:
the description is as useful as your requirement ;)